Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -75,11 +75,12 @@ public async Task<ContestDetailsServiceModel> GetContestDetails(int id)
         {
             var user = this.userProviderService.GetCurrentUser();
             var contest = await this.contestsData.GetByIdWithProblems(id);
+            var isLecturerOrAdmin = contest != null && this.lecturersInContestsBusiness.IsUserAdminOrLecturerInContest(contest!);
```

Comment:
Doesn't it make more sense to not invoke the method if contest is null since nothing "very usefull" can be returned from the method if contest is null? Only option is to return false and this is misleading information.

Reply:
I would say that if the contest is null - the user is **not** admin/lecturer in that contest and this is not unexpected/misleading - it's the correct answer. You do just that in the code with ` contest != null && this.lecturersInContestsBusiness.IsUserAdminOrLecturerInContest(contest);`.
If you think it's misleading to return `false` if the contest is `null`, then you should rewrite the logic for the variable as well, as the `var isLecturerOrAdmin` is equally misleading.

Reply:
Good point and perspective. I am making the Contest parameter for IsUserAdminOrLecturerInContest nullable and returning false if contest is null.